sia/host.go

```diff
@@ -10,8 +11,22 @@ func (c *Core) HostInfo() (components.HostInfo, error) {
 
 // TODO: Make a better UpdateHost thing.
 func (c *Core) UpdateHost(announcement components.HostAnnouncement) error {
+	info, err := c.host.HostInfo()
+	if err != nil {
+		return err
+	}
+	// TODO: This stuff should not happen here, it should be managed by
+	// hostSetConfigHandler.
+	announcementUpdate := info.Announcement
+	announcementUpdate.IPAddress = c.server.Address()
+	announcementUpdate.TotalStorage = announcement.TotalStorage
+	announcementUpdate.MaxFilesize = announcement.MaxFilesize
+	announcementUpdate.MinTolerance = announcement.MinTolerance
+	announcementUpdate.Price = announcement.Price
+	announcementUpdate.Burn = announcement.Burn
```

Comment:
Might be able to fix this right now, really exhausted though not sure if a great idea.